src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,8 +34,24 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        
+        String[] addressParts = trimmedAddress.split(",");
+        this.block = new Block(addressParts[0].trim().substring(2));
+        this.street = new Street(addressParts[1].trim());
+        if (addressParts.length > 2) {
```

Comment:
i like the way you are handling things if there are insufficient parameters. 
however this code seems to lack slap a bit, can you think of some better way to handle this?